src/reducers/RequestTypeReducer.js

```diff
@@ -0,0 +1,30 @@
+import * as Constants from '../constants/indexConstants.js'
+
+const initialState = {
+  useLegacyAddress: false,
+  receiveAddress: {},
+  uniqueLegacyAddress: false
+}
+export const requestType = (state = initialState, action) => {
+  switch (action.type) {
+    case Constants.UPDATE_RECEIVE_ADDRESS_SUCCESS:
+      let uniqueLegacy = true
+      if (action.data.receiveAddress.legacyAddress) {
+        uniqueLegacy = action.data.receiveAddress.publicAddress === action.data.receiveAddress.legacyAddress
```

Comment:
what does this do?
what is this used for?

Reply:
When on the request scene we have a smart menu at the top left. 
If the requesting wallet has a legacy address property , and that legacy address is different than the public address then the address is unique and we need to display options to choose it or the public address in the drop down . 
So, for displaying and using a Legacy address in the request screen we need to know 1. Is there a unique legacy address then we need to give the option of choosing it. If it is chosen the request screen will show the correct QR code and address